tsconfig.json

```diff
@@ -1,33 +1,20 @@
 {
-    "compilerOptions": {
-        "allowJs": true,
-        "module": "commonjs",
-        "target": "es6",
-        "outDir": "out",
-        "lib": [
-            "dom",
-            "es6"
-        ],
-        "sourceMap": true,
-        "rootDir": "src",
-        /* Strict Type-Checking Option */
-        "strict": true,   /* enable all strict type-checking options */
-        /* Additional Checks */
-        "noUnusedLocals": true, /* Report errors on unused locals. */
```

Comment:
Why are these all turned off?

Reply:
In short, we *do* want these enabled. However there's a technical block that I didn't have the time to solve yet.

Attempted explanation: We augment the existing SARIF objects (and thus types) to avoid having two nearly identical models (as the old extension had). Cleanly expressing this in the TypeScript type system (and the in-transition states) will require a pass through the codebase. Just in case there wasn't agreement in this new UI (which doesn't seem to be case now), I didn't want to do potential throw-away work.

Reply:
@roywill TypeScript strict is now enabled. I have committed the accompanying edits for conformance.

Reply:
@roywill All other flags are now at parity. Note: the diff is not 1-to-1 as `strict` by definition enables some of these other flags. Also note: `checkJs` is no longer `true` as we no longer need to `allowJs`.